taf/tests/test_updater/conftest.py

```diff
@@ -602,3 +611,53 @@ def update_target_files(target_repo: GitRepository, commit_message: str):
             new_content = existing_content + "\n" + text_to_add
             file_path.write_text(new_content, encoding="utf-8")
     target_repo.commit(commit_message)
+
+
+def update_file_without_commit(repo_path: str, filename: str):
+    text_to_add = _generate_random_text()
+    file_path = Path(repo_path) / filename
+    file_path.parent.mkdir(parents=True, exist_ok=True)  # Ensure the directory exists
+    if file_path.exists():
+        with file_path.open("a") as file:
+            file.write(text_to_add)
+    else:
+        with file_path.open("w") as file:
+            file.write(text_to_add)
+
+
+def add_file_without_commit(repo_path: str, filename: str):
+    text_to_add = _generate_random_text()
+    file_path = Path(repo_path) / filename
+    file_path.parent.mkdir(parents=True, exist_ok=True)  # Ensure the directory exists
+    with file_path.open("w") as file:
+        file.write(text_to_add)
+
+
+def remove_commits(repo_path: str, num_commits: int = 1):
+    repo = GitRepository(path=Path(repo_path))
+
+    try:
+        repo.reset_num_of_commits(num_commits, hard=True)
+    except GitError as e:
+        print(f"{str(e)}")
+
+
+def checkout_detached_head(repo_path: str):
+    """Checks out the repository to a detached HEAD state."""
+    repo = GitRepository(path=Path(repo_path))
+    head_commit_sha = repo.head_commit_sha()
+    if head_commit_sha:
+        repo.checkout_commit(head_commit_sha)
+
+
+def create_index_lock_in_repo(repo_path: str):
+    """Creates an index.lock file to simulate an interrupted git operation."""
+    repo = GitRepository(path=Path(repo_path))
+    index_lock_path = repo.path / ".git" / "index.lock"
+    index_lock_path.touch()
+
+
+def commit_file_to_repo(repo_path: str, file_name: str, commit_message: str):
```

Comment:
If you only need to commit one of the updated files, without previously adding everything that was changed to the index, you'll need to update `git.py` and `commit`. It could have optional input argument `filenames`. If they are specified, add those files to index, if not, add everything (this is how `commit` currently works

Reply:
That was from a previous implementation I don't need it now